src/server.py

```diff
@@ -239,21 +39,137 @@ def create_argparser():
+        from emulators.luis import LUISEmulator
+        return LUISEmulator()
+    elif mode.lower() == 'api':
+        from emulators.api import ApiEmulator
+        return ApiEmulator()
+    else:
+        raise ValueError("unknown mode : {0}".format(mode))
+
+
+def requires_auth(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        token = request.args.get('token', '')
+        if current_app.data_router.token is None or token == current_app.data_router.token:
+            return f(*args, **kwargs)
+        return "unauthorized", 401
+
+    return decorated
+
+
+@app.route("/parse", methods=['GET', 'POST'])
+@requires_auth
+def parse_get():
+    if request.method == 'GET':
+        request_params = request.args
+    else:
+        request_params = request.get_json(force=True)
+    if 'q' not in request_params:
+        return jsonify(error="Invalid parse parameter specified")
+    else:
+        data = current_app.data_router.extract(request_params)
+        result = current_app.data_router.parse(data["text"])
+        response = current_app.data_router.format(result)
+        return jsonify(response)
+
+
+@app.route("/status", methods=['GET'])
+@requires_auth
+def status():
+    return jsonify(current_app.data_router.get_status())
+
+
+@app.route("/", methods=['GET'])
+@requires_auth
+def hello():
+    return "hello"
+
+
+@app.route("/train", methods=['POST'])
+@requires_auth
+def train():
+    data_string = request.get_data(as_text=True)
+    current_app.data_router.start_train_proc(data_string)
+    return jsonify(info="training started with pid {0}".format(current_app.data_router.train_proc.pid))
+
+
+def setup_app(config):
     print(config.view())
-    logging.basicConfig(filename=config.log_file, level=config.log_level)
+    logging.basicConfig(filename=config['log_file'], level=config['log_level'])
     logging.captureWarnings(True)
     logging.debug(config.view())
-    try:
-        def stop(signal_number, frame):
-            raise KeyboardInterrupt()
 
-        signal.signal(signal.SIGTERM, stop)
-        server = RasaNLUServer(config)
-        server.start()
+    logging.info("Creating a new data router")
+    emulator = __create_emulator(config)
+    interpreter = __create_interpreter(config)
+    app.data_router = DataRouter(config, interpreter, emulator)
+    return app
+
 
-    except KeyboardInterrupt:
-        server.stop()
+if __name__ == '__main__':
+    # Running as standalone python application
+    parser = create_argparser()
+    cmdline_args = {key: val for key, val in vars(parser.parse_args()).items() if val is not None}
+    rasa_config = RasaNLUConfig(cmdline_args.get("config"), os.environ, cmdline_args)
+    http_server = WSGIServer(('0.0.0.0', rasa_config['port']), setup_app(rasa_config))
+    logging.info('Started http server on port %s' % rasa_config['port'])
+    http_server.serve_forever()
+
+if __name__ == 'rasa_nlu.server':
+    # Running in WSGI container
+    config_file = "config_spacy.json"
```

Comment:
so no command line args if running with `gunicorn?` is it documented how to run this way?